cpp/src/gandiva/precompiled/decimal_ops_test.cc

```diff
@@ -274,10 +330,153 @@ TEST_F(TestDecimalSql, Multiply) {
+                      false);                          // overflow
+
+  ModAndVerify(DecimalScalar128{"201", 20, 2},  // x
+               DecimalScalar128{"301", 20, 3},  // y
+               DecimalScalar128{"204", 20, 3},  // expected
+               false);                          // overflow
+
+  ModAndVerifyAllSign(DecimalScalar128{"201", 20, 3},          // x
+                      DecimalScalar128{kThirtyFive9s, 35, 2},  // y
+                      DecimalScalar128{"201", 20, 3},          // expected
+                      false);                                  // overflow
+
+  ModAndVerifyAllSign(DecimalScalar128{kThirtyFive9s, 35, 6},  // x
+                      DecimalScalar128{"201", 20, 3},          // y
+                      DecimalScalar128{"180999", 23, 6},       // expected
+                      false);                                  // overflow
+
+  ModAndVerifyAllSign(DecimalScalar128{kThirtyEight9s, 38, 20},  // x
+                      DecimalScalar128{kThirtyFive9s, 38, 21},   // y
+                      DecimalScalar128{"9990", 38, 21},          // expected
+                      false);                                    // overflow
+
+  ModAndVerifyAllSign(DecimalScalar128{"31939128063561476055", 38, 8},  // x
+                      DecimalScalar128{"10000", 20, 0},                 // y
+                      DecimalScalar128{"63561476055", 28, 8},           // expected
+                      false);
+
+  ModAndVerifyAllSign(DecimalScalar128{0, UINT64_MAX, 38, 4},  // x
+                      DecimalScalar128{0, UINT64_MAX, 38, 4},  // y
+                      DecimalScalar128{"0", 38, 4},            // expected
+                      false);                                  // overflow
+
+  ModAndVerifyAllSign(DecimalScalar128{0, UINT64_MAX, 38, 4},  // x
+                      DecimalScalar128{0, INT64_MAX, 38, 4},   // y
+                      DecimalScalar128{"1", 38, 4},            // expected
+                      false);                                  // overflow
+}
+
+TEST_F(TestDecimalSql, DivideByZero) {
+  gandiva::ExecutionContext context;
+  int32_t result_precision;
+  int32_t result_scale;
+  bool overflow;
+
+  // divide-by-zero should cause an error.
+  context.Reset();
+  result_precision = 38;
+  result_scale = 19;
+  decimalops::Divide(reinterpret_cast<int64>(&context), DecimalScalar128{"201", 20, 3},
+                     DecimalScalar128{"0", 20, 2}, result_precision, result_scale,
+                     &overflow);
+  EXPECT_EQ(context.has_error(), true);
+  EXPECT_EQ(context.get_error(), "divide by zero error");
+
+  // divide-by-nonzero should not cause an error.
+  context.Reset();
+  decimalops::Divide(reinterpret_cast<int64>(&context), DecimalScalar128{"201", 20, 3},
+                     DecimalScalar128{"1", 20, 2}, result_precision, result_scale,
+                     &overflow);
+  EXPECT_EQ(context.has_error(), false);
```

Comment:
EXPECT_TRUE and EXPECT_FALSE exists for this purpose.

Reply:
fixed